src/Common/src/System/Text/OSEncoder.cs

```diff
@@ -0,0 +1,262 @@
+                return 0;
+            
+            fixed (char *pChar = chars)
+            {
+                char *pSingleElementArray = stackalloc char[1]; // to avoid passing null to GetByteCount
+                char *pBuffer = pChar == null ?  pSingleElementArray : pChar + index;
+                return GetByteCount(pBuffer, count, flush);
+            }
+        }
+
+        public unsafe override int GetByteCount(char* chars, int count, bool flush)
+        {
+            if (chars == null)
+                throw new ArgumentNullException(nameof(chars), SR.ArgumentNull_Array);
+
+            if (count < 0)
+                throw new ArgumentOutOfRangeException(nameof(count), SR.ArgumentOutOfRange_NeedNonNegNum);
+            
+            bool excludeLastChar = count > 0 && !flush && Char.IsHighSurrogate(chars[count - 1]);
+
+            if (excludeLastChar)
+                count--;
+
+            if (_charLeftOver == NULL_CHAR)
+            {
+                if (count <= 0)
+                    return 0;
+
+                return OSEncoding.WideCharToMultiByte(_encoding.CodePage, chars, count, null, 0);
+            }
+
+            // we have left over character
+            if (count == 0 && !excludeLastChar && !flush)
+                return 0;
+
+            char [] bufferToProcess = new char[count+1];
+            bufferToProcess[0] = _charLeftOver;
+            fixed (char *pBuffer = bufferToProcess)
+            {
+                Buffer.MemoryCopy(chars, pBuffer+1, count * sizeof(char), count * sizeof(char));
+                return OSEncoding.WideCharToMultiByte(_encoding.CodePage, pBuffer, bufferToProcess.Length, null, 0);
+            }
+        }
+
+        public override unsafe int GetBytes(char[] chars, int charIndex, int charCount,
+                                              byte[] bytes, int byteIndex, bool flush)
+        {
+            if (chars == null || bytes == null)
+                throw new ArgumentNullException(chars == null ? nameof(chars) : nameof(bytes), SR.ArgumentNull_Array);
+
+            if (charIndex < 0 || charCount < 0)
+                throw new ArgumentOutOfRangeException(charIndex < 0 ? nameof(charIndex) : nameof(charCount), SR.ArgumentOutOfRange_NeedNonNegNum);
+
+            if (chars.Length - charIndex < charCount)
+                throw new ArgumentOutOfRangeException(nameof(chars), SR.ArgumentOutOfRange_IndexCountBuffer);
+
+            if (byteIndex < 0 || byteIndex > bytes.Length)
+                throw new ArgumentOutOfRangeException(nameof(byteIndex), SR.ArgumentOutOfRange_Index);
+            
+            if (bytes.Length == 0)
+                return 0;
+
+            if (charCount == 0 && (_charLeftOver == NULL_CHAR || !flush))
+                return 0;
+            
+            fixed (char* pChars = chars)
+            fixed (byte* pBytes = bytes)
+            {
+                char *pSingleElementArray = stackalloc char[1]; // to avoid passing null to GetByteCount
```

Comment:
I'll change that

---
In reply to: [93727715](https://github.com/dotnet/corefx/pull/14649#discussion_r93727715) [](ancestors = 93727715)